sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java

```diff
@@ -202,6 +209,7 @@ private ShardId shardId(String indexName, String id, @Nullable String routing) {
             ).shardId();
         } catch (IndexMissingException e) {
             if (!autoCreateIndices) {
+                indicesDeleted.add(indexName);
```

Comment:
index re-creation must be prevented while adding new items. so this set will be checked on add(..)